### DISCUSSION TITLE
Offline install - CS_Tools

### DISCUSSION THREAD

**Discussion by sivaprasadsn** [thoughtspot/cs_tools#103] [General]
I am trying to install cs_tools offline, but hit by bunch of issues. Trying to fine right way to do it. Please advice

[admin@ash-tspotfindv01 cs_tools-master]$ python -m sysconfig
Platform: "linux-x86_64"
Python version: "3.9"
Current installation scheme: "posix_prefix"

Paths:
	data = "/usr"
	include = "/usr/include/python3.9"
	platinclude = "/usr/include/python3.9"
	platlib = "/usr/lib64/python3.9/site-packages"
	platstdlib = "/usr/lib64/python3.9"
	purelib = "/usr/lib/python3.9/site-packages"
	scripts = "/usr/bin"
	stdlib = "/usr/lib64/python3.9"

[admin@ash-tspotfindv01 cs_tools-master]$ python3 -m sysconfig | more
Platform: "linux-x86_64"
Python version: "3.9"
Current installation scheme: "posix_prefix"

Paths:
	data = "/usr"
	include = "/usr/include/python3.9"
	platinclude = "/usr/include/python3.9"
	platlib = "/usr/lib64/python3.9/site-packages"
	platstdlib = "/usr/lib64/python3.9"
	purelib = "/usr/lib/python3.9/site-packages"
	scripts = "/usr/bin"
	stdlib = "/usr/lib64/python3.9"

**Comment by boonhapus** [comment]
Hi @sivaprasadsn ! Happy to help. It looks like you might've downloaded the repo (`cs_tools-master`). We have an installer path to follow instead of running from source. Can you try the steps below?

1. Extract the zip file.
2. Run python against the `_bootstrapper.py` file with `--offline-mode` specified

`cd cs-tools_1.4.16_linux-x86_64_3.9`
`python _bootstrapper.py --install --offline-mode .`

It's a bit too big for GitHub discussions, so I've attached an Egnyte link. Let me know if that works for you!
**[https://thoughtspot.egnyte.com/dl/RdF3OIxjBK/cs-tools_v1.4.16_linux-x86_64_3.9.zip](https://thoughtspot.egnyte.com/dl/RdF3OIxjBK/cs-tools_v1.4.16_linux-x86_64_3.9.zip_)**

**Comment by sivaprasadsn** [comment]
[cs_tools-bootstrap-error-tmpssc5e0oz.log](https://github.com/thoughtspot/cs_tools/files/14341196/cs_tools-bootstrap-error-tmpssc5e0oz.log)
Please find installation log here with attached, please review and advice. 
I see installation failing with error,
